File: 190. Reverse Bits/main.py
class Solution:
    def reverseBits(self, n: int) -> int:
        result = 0

        for i in range(32):
            result <<= 1
            result += n & 1
            n >>= 1

        return result


class Solution:
    def reverseBits(self, n: int) -> int:
        return int(bin(n)[2:].zfill(32)[::-1], 2)

File: 190. Reverse Bits/main.rs
struct Solution;

impl Solution {
    pub fn reverse_bits(x: u32) -> u32 {
        let mut result = 0;
        let mut n = x;

        for _ in 0..32 {
            result <<= 1;
            result += n & 1;
            n >>= 1;
        }

        result
    }
}

impl Solution {
    pub fn reverse_bits(x: u32) -> u32 {
        let mut result = 0;

        for i in 0..32 {
            result = (result << 1) + ((x >> i) & 1);
        }

        result
    }
}

impl Solution {
    pub fn reverse_bits(x: u32) -> u32 {
        x.reverse_bits()
    }
}
